File: backend/rails/app/models/token.rb
# Token de acesso simples (Bearer) usado pela API — ver
# app/controllers/api/auth_controller.rb e app/controllers/api/base_controller.rb.
class Token < ApplicationRecord
  belongs_to :user

  before_validation { self.key = SecureRandom.hex(32) if key.blank? }

  validates :key, presence: true, uniqueness: true
end

File: backend/laravel/app/Models/Token.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Support\Str;

/**
 * Token de acesso simples (Bearer) usado pela API — ver AuthController.
 */
class Token extends Model
{
    protected $fillable = ['user_id', 'key'];

    public $timestamps = false;

    protected function casts(): array
    {
        return [
            'created_at' => 'datetime',
        ];
    }

    protected static function booted(): void
    {
        static::creating(function (Token $token) {
            if (! $token->key) {
                $token->key = Str::random(64);
            }
        });
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}
